README.md

```diff
@@ -732,6 +732,30 @@ from an object:
 ...     Object.metadata.apply_atomic_operations( *[AVUOperation(operation='remove', avu=i) for i in avus_on_Object] )
 ```
 
+Extracting JSON encoded server information in case of error
+-----------------------------------------------------------
+
+Some server apis, including atomic metadata and replica truncation, can fail for various reasons and generate an
+exception.  In these cases the message object returned from the server is made available in the 'server_msg' attribute
+of the iRODSException object.
+
+This enables an approach like the following, which logs server information possibly underlying the error that was evoked:
+
+```python
+    try:
+        Object.metadata.apply_atomic_operations( ops )
+        # or:
+        DataObject.replica_truncate( size )
+    except iRODSException as exc:
+        log.error('Server API call failure. Traceback = %r ; iRODS Server info = %r',
+            traceback.extract_tb(sys.exc_info()[2]),
+            exc.server_msg.get_json_encoded_struct())
+```
+
+For `DataObject.replica_truncate(...)`, note that exc.server_msg.get_json_encoded_struct() can be used in the exception-handling
+code path to retrieve the same information that would have been routinely returned from the truncate call itself, had it actually
+completed without error.
```

Comment:
More the latter.  For whatever reason, atomic metadata the endpoint in the python client isn't a JSON string, which maybe I should look into.

Reply:
Looks like during a routine successful call of atomic metadata, iRODS returns an empty JSON structure, { }.  I think in replica_truncate, we're more oriented toward informative output , due to the complexity of the interface probably.  Atomic metadata calls tend to succeed, even when adding preexisting AVUs or removing ones that don't exist, which I think is part of the design.

Reply:
Okay, that makes sense. I only brought this up because it wasn't clear whether this comment was specific to the API. Maybe we can generalize the language slightly so that it is clear that any API which returns JSON in non-error cases can be used in this way (you can even say "for example, `DataObject.replica_truncate(...)`").

Reply:
Think we can resolve this?

Reply:
I do not see any updated wording, but if everybody is okay with how it is now, I'm happy to resolve this.

Reply:
Ok, I'm happy and I think Kory and any others who have an opinion have seen this, so resolving....